## Mandatory Kaggle Stuff

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/digit-recognizer/sample_submission.csv
/kaggle/input/digit-recognizer/train.csv
/kaggle/input/digit-recognizer/test.csv
/kaggle/input/full-and-finalproject/submition.csv
/kaggle/input/full-and-finalproject/__results__.html
/kaggle/input/full-and-finalproject/__notebook_source__.ipynb
/kaggle/input/full-and-finalproject/filter5x5inc.csv
/kaggle/input/full-and-finalproject/filter9x9.csv
/kaggle/input/full-and-finalproject/filter7x7simple.csv
/kaggle/input/full-and-finalproject/filter9x9inc.csv
/kaggle/input/full-and-finalproject/__notebook__.ipynb
/kaggle/input/full-and-finalproject/filter5x5simple.csv
/kaggle/input/full-and-finalproject/__output__.json
/kaggle/input/full-and-finalproject/filter7x7inc.csv
/kaggle/input/full-and-finalproject/custom.css


## Importing Libraries

In [2]:
import pandas as pd
import scipy
from scipy import signal
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.naive_bayes import MultinomialNB
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics
from sklearn.tree import DecisionTreeClassifier
from itertools import chain
pd.set_option('display.max_colwidth', 500)


In [3]:
#Importing train data
df = pd.read_csv('../input/digit-recognizer/train.csv')

In [4]:
#Creating X & Y for Train_test_split
X = df.drop(["label"], axis=1)
Y = df["label"]

## Making filters

In [5]:
filter9 = np.array([[1 for x in range(9)] for y in range(9)])
filter7 = np.array([[1 for x in range(7)] for y in range(7)])
filter5 = np.array([[1 for x in range(5)] for y in range(5)])

filter9inc = np.array([[1,1,1,1,1,1,1,1,1],
                      [1,2,2,2,2,2,2,2,1],
                      [1,2,3,3,3,3,3,2,1],
                      [1,2,3,4,4,4,3,2,1],
                      [1,2,3,4,5,4,3,2,1],
                      [1,2,3,4,4,4,3,2,1],
                      [1,2,3,3,3,3,3,2,1],
                      [1,2,2,2,2,2,2,2,1],
                      [1,1,1,1,1,1,1,1,1]])

filter7inc = np.array([[1,1,1,1,1,1,1],
                      [1,2,2,2,2,2,1],
                      [1,2,3,3,3,2,1],
                      [1,2,3,4,3,2,1],
                      [1,2,3,3,3,2,1],
                      [1,2,2,2,2,2,1],
                      [1,1,1,1,1,1,1]])

filter5inc = np.array([[1,1,1,1,1],
                      [1,2,2,2,1],
                      [1,2,3,2,1],
                      [1,2,2,2,1],
                      [1,1,1,1,1]])

In [6]:
print(filter9.shape,filter9inc.shape)
print(filter7.shape,filter7inc.shape)
print(filter5.shape,filter5inc.shape)

(9, 9) (9, 9)
(7, 7) (7, 7)
(5, 5) (5, 5)


## Applying Convolutions using 6 filters

### 5x5 Filters

In [7]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter5, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [8]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter5x5simple.csv")

(42000, 576)
(42000,)


In [9]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter5inc, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [10]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter5x5inc.csv")

(42000, 576)
(42000,)


### 7x7 Filters

In [11]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter7, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [12]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter7x7simple.csv")

(42000, 484)
(42000,)


In [13]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter7inc, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [14]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter7x7inc.csv")

(42000, 484)
(42000,)


### 9x9 Filters

In [15]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter9, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [16]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter9x9.csv")

(42000, 400)
(42000,)


In [17]:
df=pd.DataFrame()
arr = np.asarray(X)
for i in range(0,42000):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter9inc, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==30000 or i==40000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

10000
20000
30000
40000


In [18]:
#Cheking the data shape and Saving the data
print(df.shape)
print(Y.shape)
df.to_csv("./filter9x9inc.csv")

(42000, 400)
(42000,)


## Now using the New(Convoluted) Datasets

In [19]:
#Importing the dataset and deleting the extra column
df = pd.read_csv('../input/digit-recognizer/train.csv')
df5=pd.read_csv('../input/full-and-finalproject/filter5x5simple.csv')
del df5['Unnamed: 0']
df5inc=pd.read_csv('../input/full-and-finalproject/filter5x5inc.csv')
del df5inc['Unnamed: 0']
df7=pd.read_csv('../input/full-and-finalproject/filter7x7simple.csv')
del df7['Unnamed: 0']
df7inc=pd.read_csv('../input/full-and-finalproject/filter7x7inc.csv')
del df7inc['Unnamed: 0']
df9=pd.read_csv('../input/full-and-finalproject/filter9x9.csv')
del df9['Unnamed: 0']
df9inc=pd.read_csv('../input/full-and-finalproject/filter9x9inc.csv')
del df9inc['Unnamed: 0']
#making the dataframe for accuracy
dfacc=pd.DataFrame(columns=["FilterSize","Model Used","Accuracy"])


In [20]:
#method to insert acc in dataframe
def insertAcc(dff,filters,model,acc):
    lst=[filters,model,acc]
    dff.loc[len(dff)] = lst

## Methods To be used on all 6 Datasets

In [21]:
y = df["label"]
meanScores=[]
def predOnDf(dataf,y,fSzie,dfacc):
    X = dataf #Creating X for Train_test_split
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42) #spliting
    # ***************************** Multinomial NB *****************************
    mnb = MultinomialNB()
    mnb.fit(X_train, y_train)
    mnbYPred = mnb.predict(X_test)
    mnbAcc = metrics.accuracy_score(y_test, mnbYPred)
    print ("Naive Bayes Accuracy: ", mnbAcc*100)
    insertAcc(dfacc,fSzie,"Naive Bayes",mnbAcc*100) #inserting accuracy into dataframe
    nb_scores = cross_val_score(mnb, X_train, y_train, cv=6)
    print('Naive Bayes Scores: ',nb_scores)
    # ***************************** SVM *****************************
    svmModel = SVC(kernel="rbf", random_state=42, verbose=True,C=9)
    svmModel.fit(X_train, y_train)
    svmYpred = svmModel.predict(X_test)
    svmAcc = metrics.accuracy_score(y_test,svmYpred)
    print('SVM Accuracy: ', svmAcc*100)
    insertAcc(dfacc,fSzie,"SVM",svmAcc*100) #inserting accuracy into dataframe
    svc_scores = cross_val_score(svmModel, X_train, y_train, cv=6)
    print('SVC Scores: ',svc_scores)
    # ***************************** Linear Regression *****************************
    reg = LinearRegression()
    reg.fit(X_train, y_train)
    regYpred = reg.predict(X_test)
    regAcc = metrics.r2_score(y_test,regYpred) 
    print('Linear Regression Accuracy: ', regAcc*100)
    insertAcc(dfacc,fSzie,"Linear Regression",regAcc*100)#inserting accuracy into dataframe
    reg_scores = cross_val_score(reg, X_train, y_train, cv=6)
    print('Reg Scores: ',reg_scores)
    # *****************************  KNN *****************************
    KNNmodel = KNeighborsClassifier()
    KNNmodel.fit(X_train, y_train)
    knnYpred = KNNmodel.predict(X_test)
    knnAcc = metrics.accuracy_score(y_test, knnYpred)
    print('KNN Accuracy: ',knnAcc*100)
    insertAcc(dfacc,fSzie,"KNN",knnAcc*100)#inserting accuracy into dataframe
    knn_scores = cross_val_score(KNNmodel, X_train, y_train, cv=6)
    print('KNN Scores: ',knn_scores)
    #MeanScores
    meanScores.append(reg_scores.mean())
    meanScores.append(nb_scores.mean())
    meanScores.append(svc_scores.mean())
    meanScores.append(knn_scores.mean())
    print(dfacc)

## Making Predictions

### For 5x5 Datasets

In [22]:
predOnDf(df5,y,"5x5simple",dfacc)

Naive Bayes Accuracy:  76.14285714285714
Naive Bayes Scores:  [0.77047619 0.75634921 0.76301587 0.76825397 0.76396825 0.77095238]
[LibSVM]SVM Accuracy:  97.90476190476191
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.97968254 0.98111111 0.97619048 0.98063492 0.97698413 0.97888889]
Linear Regression Accuracy:  60.54374415007434
Reg Scores:  [ 6.11267232e-01 -4.22540300e+13  6.05579727e-01  6.06777387e-01
  5.81405968e-01  5.96007139e-01]
KNN Accuracy:  96.61904761904762
KNN Scores:  [0.96698413 0.96238095 0.9652381  0.96793651 0.9652381  0.9684127 ]
  FilterSize         Model Used   Accuracy
0  5x5simple        Naive Bayes  76.142857
1  5x5simple                SVM  97.904762
2  5x5simple  Linear Regression  60.543744
3  5x5simple                KNN  96.619048


In [23]:
predOnDf(df5inc,y,"5x5inc",dfacc)

Naive Bayes Accuracy:  77.14285714285715
Naive Bayes Scores:  [0.78079365 0.76888889 0.77587302 0.77825397 0.77174603 0.7831746 ]
[LibSVM]SVM Accuracy:  98.11904761904762
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.98079365 0.98126984 0.97857143 0.98111111 0.97904762 0.98142857]
Linear Regression Accuracy:  60.76247887043299
Reg Scores:  [0.61294836 0.59721664 0.61514884 0.61051926 0.60556645 0.59838609]
KNN Accuracy:  96.78571428571429
KNN Scores:  [0.97095238 0.96714286 0.96777778 0.97238095 0.9684127  0.97047619]
  FilterSize         Model Used   Accuracy
0  5x5simple        Naive Bayes  76.142857
1  5x5simple                SVM  97.904762
2  5x5simple  Linear Regression  60.543744
3  5x5simple                KNN  96.619048
4     5x5inc        Naive Bayes  77.142857
5     5x5inc                SVM  98.119048
6     5x5inc  Linear Regression  60.762479
7     5x5inc                KNN  96.785714


### For 7x7 Datasets

In [24]:
predOnDf(df7,y,"7x7simple",dfacc)

Naive Bayes Accuracy:  71.02380952380952
Naive Bayes Scores:  [0.71952381 0.70650794 0.70936508 0.72031746 0.7115873  0.71888889]
[LibSVM]SVM Accuracy:  97.4047619047619
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.97460317 0.97349206 0.97095238 0.9747619  0.97222222 0.97571429]
Linear Regression Accuracy:  60.49792885212695
Reg Scores:  [0.61116288 0.59944854 0.61195009 0.60928304 0.60598563 0.59680395]
KNN Accuracy:  94.88095238095238
KNN Scores:  [0.95174603 0.94079365 0.94777778 0.9484127  0.94571429 0.95126984]
   FilterSize         Model Used   Accuracy
0   5x5simple        Naive Bayes  76.142857
1   5x5simple                SVM  97.904762
2   5x5simple  Linear Regression  60.543744
3   5x5simple                KNN  96.619048
4      5x5inc        Naive Bayes  77.142857
5      5x5inc                SVM  98.119048
6      5x5inc  Linear Regression  60.762479
7      5x5inc                KNN  96.785714
8   7x7simple        Naive Bayes  71.023810
9   7x7simple      

In [25]:
predOnDf(df7inc,y,"7x7inc",dfacc)

Naive Bayes Accuracy:  73.33333333333333
Naive Bayes Scores:  [0.74206349 0.72936508 0.73666667 0.74301587 0.73746032 0.74587302]
[LibSVM]SVM Accuracy:  97.73809523809524
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.97650794 0.97984127 0.97587302 0.97904762 0.97825397 0.97714286]
Linear Regression Accuracy:  60.60312330604967
Reg Scores:  [0.61059663 0.59836508 0.61194094 0.60799098 0.60358018 0.59583541]
KNN Accuracy:  95.88095238095238
KNN Scores:  [0.95809524 0.95444444 0.95936508 0.95904762 0.9584127  0.96      ]
   FilterSize         Model Used   Accuracy
0   5x5simple        Naive Bayes  76.142857
1   5x5simple                SVM  97.904762
2   5x5simple  Linear Regression  60.543744
3   5x5simple                KNN  96.619048
4      5x5inc        Naive Bayes  77.142857
5      5x5inc                SVM  98.119048
6      5x5inc  Linear Regression  60.762479
7      5x5inc                KNN  96.785714
8   7x7simple        Naive Bayes  71.023810
9   7x7simple     

### For 9x9 Dataset

In [26]:
predOnDf(df9,y,"9x9simple",dfacc)

Naive Bayes Accuracy:  67.23809523809524
Naive Bayes Scores:  [0.68349206 0.67206349 0.6715873  0.68095238 0.68063492 0.68238095]
[LibSVM]SVM Accuracy:  96.30952380952381
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.96730159 0.96650794 0.96444444 0.97047619 0.96301587 0.96809524]
Linear Regression Accuracy:  -154707922974569.12
Reg Scores:  [ 5.80077570e-01 -1.43201009e+11  5.88099953e-01  5.84702382e-01
 -1.90951196e+10  5.70299132e-01]
KNN Accuracy:  92.54761904761905
KNN Scores:  [0.92571429 0.92142857 0.92603175 0.92079365 0.92111111 0.92857143]
   FilterSize         Model Used      Accuracy
0   5x5simple        Naive Bayes  7.614286e+01
1   5x5simple                SVM  9.790476e+01
2   5x5simple  Linear Regression  6.054374e+01
3   5x5simple                KNN  9.661905e+01
4      5x5inc        Naive Bayes  7.714286e+01
5      5x5inc                SVM  9.811905e+01
6      5x5inc  Linear Regression  6.076248e+01
7      5x5inc                KNN  9.678571e+01
8 

In [27]:
predOnDf(df9inc,y,"9x9inc",dfacc)

Naive Bayes Accuracy:  70.38095238095238
Naive Bayes Scores:  [0.71380952 0.69904762 0.70730159 0.71206349 0.70904762 0.71333333]
[LibSVM]SVM Accuracy:  97.14285714285714
[LibSVM][LibSVM][LibSVM][LibSVM][LibSVM][LibSVM]SVC Scores:  [0.97253968 0.97285714 0.97079365 0.9747619  0.97238095 0.97365079]
Linear Regression Accuracy:  60.35209906710299
Reg Scores:  [0.60667466 0.59376053 0.60774287 0.60560832 0.60504998 0.59386343]
KNN Accuracy:  94.35714285714286
KNN Scores:  [0.94285714 0.93555556 0.94063492 0.94142857 0.93968254 0.9447619 ]
   FilterSize         Model Used      Accuracy
0   5x5simple        Naive Bayes  7.614286e+01
1   5x5simple                SVM  9.790476e+01
2   5x5simple  Linear Regression  6.054374e+01
3   5x5simple                KNN  9.661905e+01
4      5x5inc        Naive Bayes  7.714286e+01
5      5x5inc                SVM  9.811905e+01
6      5x5inc  Linear Regression  6.076248e+01
7      5x5inc                KNN  9.678571e+01
8   7x7simple        Naive Bayes  7

### Cross Validation

In [28]:
meanScores=np.array_split(meanScores, 6)
nb=[meanScores[0][0],meanScores[1][0],meanScores[2][0],meanScores[3][0],meanScores[4][0],meanScores[5][0]]
svm=[meanScores[0][1],meanScores[1][1],meanScores[2][1],meanScores[3][1],meanScores[4][1],meanScores[5][1]]
lr=[meanScores[0][2],meanScores[1][2],meanScores[2][2],meanScores[3][2],meanScores[4][2],meanScores[5][2]]
knn=[meanScores[0][3],meanScores[1][3],meanScores[2][3],meanScores[3][3],meanScores[4][3],meanScores[5][3]]

models=['Naive Bayes','SVM','Linear Regression','KNN']
#df of cross calidation scores
cv = {'Size': [ '5x5', '5x5inc','7x7','7x7inc','9x9','9x9inc'], 
    models[0]: nb,models[1]: svm,models[2]: lr,models[3]: knn}

cv_df = pd.DataFrame(cv)
cv_df

Size   Naive Bayes       SVM  Linear Regression       KNN
0     5x5 -7.042338e+12  0.765503           0.978915  0.966032
1  5x5inc  6.066309e-01  0.776455           0.980370  0.969524
2     7x7  6.057724e-01  0.714365           0.973624  0.947619
3  7x7inc  6.047182e-01  0.739074           0.977778  0.958228
4     9x9 -2.704935e+10  0.678519           0.966640  0.923942
5  9x9inc  6.021166e-01  0.709101           0.972831  0.940820

In [29]:
#checking Which combination has the highest Accuracy
#dfacc.columns
dfacc.sort_values('Accuracy',ascending=False)

FilterSize         Model Used      Accuracy
5      5x5inc                SVM  9.811905e+01
1   5x5simple                SVM  9.790476e+01
13     7x7inc                SVM  9.773810e+01
9   7x7simple                SVM  9.740476e+01
21     9x9inc                SVM  9.714286e+01
7      5x5inc                KNN  9.678571e+01
3   5x5simple                KNN  9.661905e+01
17  9x9simple                SVM  9.630952e+01
15     7x7inc                KNN  9.588095e+01
11  7x7simple                KNN  9.488095e+01
23     9x9inc                KNN  9.435714e+01
19  9x9simple                KNN  9.254762e+01
4      5x5inc        Naive Bayes  7.714286e+01
0   5x5simple        Naive Bayes  7.614286e+01
12     7x7inc        Naive Bayes  7.333333e+01
8   7x7simple        Naive Bayes  7.102381e+01
20     9x9inc        Naive Bayes  7.038095e+01
16  9x9simple        Naive Bayes  6.723810e+01
6      5x5inc  Linear Regression  6.076248e+01
14     7x7inc  Linear Regression  6.060312e+01
2   5x5simple  Linear Regression  6.054374e+01
10  7x7simple  Linear Regression  6.049793e+01
22     9x9inc  Linear Regression  6.035210e+01
18  9x9simple  Linear Regression -1.547079e+14

## Highest Accurcy: 98.11% (Svm,5x5inc)

### Training the model for final Predictions

In [30]:
X = df5inc #Creating X for Train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42) #spliting
# ***************************** SVM *****************************
svmModel = SVC(kernel="rbf", random_state=42, verbose=True,C=9)
svmModel.fit(X_train, y_train)

[LibSVM]

SVC(C=9, random_state=42, verbose=True)

### Reducing the features with the same filter as used for training

In [31]:
test=pd.read_csv("../input/digit-recognizer/test.csv")
df=pd.DataFrame()
arr = np.asarray(test)
for i in range(len(test)):  #Looping through Every row
    x = None  #Nullifying the variables for every iteration
    ax = None
    xRes = None
    array_1d = None
    
    xRes = np.reshape(arr[i],(-1,28))  #reshaping the data to image size
    x = scipy.signal.convolve2d(xRes, filter5inc, mode='valid') #Applying convolution with one of the two Filters 
    if i == 10000 or i==20000 or i==5000 or i==15000 or i==25000:     # Printing the iteration value to check the progress
        print(i)
    ax = x
    array_1d = ax.flatten()   #flattening the convolve data to 1d array
    df = df.append(pd.DataFrame(array_1d).T, ignore_index=True) #appending it to the new dataframe

5000
10000
15000
20000
25000


### Making Predictions And saving the file

In [32]:
svmFinalpred=svmModel.predict(df)
finalPred=pd.DataFrame(svmFinalpred,columns=["Label"])
finalPred['ImageId']=finalPred.index+1
finalPred = finalPred.reindex(['ImageId','Label'], axis=1)
finalPred.to_csv('./submition.csv',index=False)

## Finally I got 0.98082 Score on Kaggle.